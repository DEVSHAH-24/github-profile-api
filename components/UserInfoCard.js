import React from "react";
import { Text, Image } from "react-native";
import { View } from "react-native";
import { reactLogo } from "../assets/Constants";
import { appStyles } from "../styles/appStyles";

export function UserInfoCard({ userData }) {
  return (
    <View style={appStyles.userContainer}>
      <Image
        style={appStyles.stretch}
        source={{
          uri:
            userData?.avatar_url === undefined
              ? reactLogo
              : userData.avatar_url,
        }}
      />
      {userData?.name === undefined ? (
        <Text style={appStyles.textStyle}>Name</Text>
      ) : userData.name != null ? (
        <Text style={appStyles.textStyle}>{userData.name}</Text>
      ) : (
        <Text
          style={[
            appStyles.textStyle,
            { fontStyle: "italic", fontWeight: "200" },
          ]}
        >
          Name not available
        </Text>
      )}

      {userData?.location === undefined ? (
        <Text style={appStyles.subheadingTextStyle}>
          Location will appear here
        </Text>
      ) : userData.location !== null ? (
        <Text style={appStyles.subheadingTextStyle}>{userData.location}</Text>
      ) : null}

      <Text style={appStyles.subheadingTextStyle}>
        {userData?.login !== undefined
          ? userData.hireable !== true
            ? "Not hireable"
            : "Hireable"
          : ""}
      </Text>
    </View>
  );
}

export default UserInfoCard;
