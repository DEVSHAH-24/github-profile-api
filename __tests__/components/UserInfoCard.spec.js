import * as React from "react";
import TestRenderer from "react-test-renderer";
import UserInfoCard from "../../components/UserInfoCard";

const getComponent = (props = {}) => <UserInfoCard {...props} />;

describe("UserInfoCard", () => {
  test("Snapshot", () => {
    const component = getComponent();

    const testRenderer = TestRenderer.create(component);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  test("Snapshot with Data", () => {
    const component = getComponent({
      userData: {
        login: "debajitdeb11",
        id: 11337871,
        node_id: "MDQ6VXNlcjExMzM3ODcx",
        avatar_url: "https://avatars.githubusercontent.com/u/11337871?v=4",
        gravatar_id: "",
        url: "https://api.github.com/users/debajitdeb11",
        html_url: "https://github.com/debajitdeb11",
        followers_url: "https://api.github.com/users/debajitdeb11/followers",
        following_url:
          "https://api.github.com/users/debajitdeb11/following{/other_user}",
        gists_url: "https://api.github.com/users/debajitdeb11/gists{/gist_id}",
        starred_url:
          "https://api.github.com/users/debajitdeb11/starred{/owner}{/repo}",
        subscriptions_url:
          "https://api.github.com/users/debajitdeb11/subscriptions",
        organizations_url: "https://api.github.com/users/debajitdeb11/orgs",
        repos_url: "https://api.github.com/users/debajitdeb11/repos",
        events_url:
          "https://api.github.com/users/debajitdeb11/events{/privacy}",
        received_events_url:
          "https://api.github.com/users/debajitdeb11/received_events",
        type: "User",
        site_admin: false,
        name: "DEBAJIT DEB",
        company: "OnSolve",
        blog: "",
        location: "India",
        email: null,
        hireable: true,
        bio: '"you\'re gonna die unknown, do something about it"',
        twitter_username: null,
        public_repos: 43,
        public_gists: 4,
        followers: 6,
        following: 17,
        created_at: "2015-03-05T19:08:24Z",
        updated_at: "2022-09-15T17:40:12Z",
      },
    });

    const testRenderer = TestRenderer.create(component);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  test("Snapshot with some data", () => {
    const component = getComponent({
      userData: {
        login: "debajitdeb11",
        id: 11337871,
        node_id: "MDQ6VXNlcjExMzM3ODcx",
        avatar_url: null,
        gravatar_id: "",
        url: "https://api.github.com/users/debajitdeb11",
        html_url: "https://github.com/debajitdeb11",
        followers_url: "https://api.github.com/users/debajitdeb11/followers",
        following_url:
          "https://api.github.com/users/debajitdeb11/following{/other_user}",
        gists_url: "https://api.github.com/users/debajitdeb11/gists{/gist_id}",
        starred_url:
          "https://api.github.com/users/debajitdeb11/starred{/owner}{/repo}",
        subscriptions_url:
          "https://api.github.com/users/debajitdeb11/subscriptions",
        organizations_url: "https://api.github.com/users/debajitdeb11/orgs",
        repos_url: "https://api.github.com/users/debajitdeb11/repos",
        events_url:
          "https://api.github.com/users/debajitdeb11/events{/privacy}",
        received_events_url:
          "https://api.github.com/users/debajitdeb11/received_events",
        type: "User",
        site_admin: false,
        name: null,
        company: "OnSolve",
        blog: "",
        location: null,
        email: null,
        hireable: false,
        bio: '"you\'re gonna die unknown, do something about it"',
        twitter_username: null,
        public_repos: 43,
        public_gists: 4,
        followers: 6,
        following: 17,
        created_at: "2015-03-05T19:08:24Z",
        updated_at: "2022-09-15T17:40:12Z",
      },
    });

    const testRenderer = TestRenderer.create(component);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
