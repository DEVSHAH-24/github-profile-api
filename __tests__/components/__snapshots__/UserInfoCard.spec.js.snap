// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoCard Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "padding": 35,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactnative.dev/img/tiny_logo.png",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
        "paddingVertical": 20,
      }
    }
  >
    Name
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "400",
        "paddingVertical": 5,
      }
    }
  >
    Location will appear here
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "400",
        "paddingVertical": 5,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`UserInfoCard Snapshot with Data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "padding": 35,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://avatars.githubusercontent.com/u/11337871?v=4",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
        "paddingVertical": 20,
      }
    }
  >
    DEBAJIT DEB
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "400",
        "paddingVertical": 5,
      }
    }
  >
    India
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "400",
        "paddingVertical": 5,
      }
    }
  >
    Hireable
  </Text>
</View>
`;

exports[`UserInfoCard Snapshot with some data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "padding": 35,
    }
  }
>
  <Image
    source={
      Object {
        "uri": null,
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 25,
          "fontWeight": "700",
          "paddingVertical": 20,
        },
        Object {
          "fontStyle": "italic",
          "fontWeight": "200",
        },
      ]
    }
  >
    Name not available
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "400",
        "paddingVertical": 5,
      }
    }
  >
    Not hireable
  </Text>
</View>
`;
